Add encryption option and a cleaner cli interface

andrewrynhard opened the pull request:
This PR brings quite a bit of changes. Instead of using the standard flags, `https://github.com/codegangsta/cli.git` was implemented to use subcommands, global flags, and subcommand flags. Additionally an encryption option was added that provides encryption on a per-task basis. This is useful for things like ssh keys. Storing your ssh keys and other sensitive files in your github repo can now be secure and allows a user to keep all things under a common repo. This makes bootstrapping/deploying much easier.

Commit:
Add encryption and other changes

This commit adds the ability to encrypt sensitive files. Additionally the cli
package by codegangsta is used for a more intuitive usage.

Commit:
Rename 'Encs' to 'Secrets'

FooSoft commented:
Thank you for the PR, I'll be looking over it in detail over the next couple of days (I've actually seriously considered adding encryption in the past but never gotten around to it). 

The new CLI seems pretty slick but I'm a little worried about breaking the shell scripts people currently have for automating homemaker stuff. Speaking of automation, another thing that might be nice is the ability to provide an encryption password on the command line so that it can be executed without user interaction.

andrewrynhard commented:
@FooSoft: Yes anything that could break the shell scripts is definitely not desirable. That being said, if homemaker is to become anything more than what it is now (more features) then IMHO the cli will have to be more robust. On the plus side the change they will have to make will be minimal because homemake handles a lot of the work anyways. The standard golang flags just makes things harder (readability especially) and doesn't adhere to the small convention of using `--` rather than `-` for flags. Very minor but still bugs me :). If there will be any braking changes better they happen now while the project is still young.

As for the option to pass in a password, that would be nice. I can mix that in. I would like to also keep the interactive version so I don't have to store my password in a shell script. Does keeping both options around sound good?

Commit:
Add features

A password can now be provided as a flag. Another command was added to decrypt
any secrets.

Commit:
Fix decrypt message

Commit:
Fix decrypt cli alias

Commit:
Cleanup code

FooSoft commented:
Yup, both options sound good and they are in-line with what Homemaker does anyway for things like `clobber` and `force` (it asks what to do if these options were not provided on the command line). 

You sold me on the CLI change too, will just have to update `README.md` before this can go to master.
